globus_automate_client/action_client.py

```diff
@@ -118,54 +112,30 @@ def release(self, action_id: str) -> GlobusHTTPResponse:
         path = self.qjoin_path(action_id, "release")
         return self.post(path)
 
+    @classmethod
+    def new_client(
+        cls, action_url: str, authorizer: GlobusAuthorizer
```

Comment:
Elsewhere we've stated:
```
    allowed_authorizer_types = (
        AccessTokenAuthorizer,
        RefreshTokenAuthorizer,
        ClientCredentialsAuthorizer,
    )
```
which is consumed by the `BaseClient`. If we can make the declaration of `authorizer` also support that, then we get the type system supporting what the run-time does. I think a `Union` would do the trick?